pkg/auth/oauth2/auth_openshift.go

```diff
@@ -229,12 +232,85 @@ func (o *openShiftAuth) Authenticate(_ http.ResponseWriter, r *http.Request) (*a
 	if err != nil {
 		return nil, err
 	}
+
 	if cookie.Value == "" {
 		return nil, fmt.Errorf("unauthenticated, no value for cookie %s", sessions.OpenshiftAccessTokenCookieName)
 	}
 
+	tokenReviewURL, err := url.Parse(o.issuerURL)
+	if err != nil {
+		klog.Errorf("failed to parse the URL to kube-apiserver: %v", err)
+		return nil, err
+	}
+	tokenReviewURL.Path = "/apis/authentication.k8s.io/v1/tokenreviews"
+	tokenReviewURL.RawPath = "/apis/authentication.k8s.io/v1/tokenreviews"
+
+	tokenReview := &authv1.TokenReview{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "authentication.k8s.io/v1",
+			Kind:       "TokenReview",
+		},
+		Spec: authv1.TokenReviewSpec{
+			Token: cookie.Value,
+		},
+	}
+
+	tokenReviewJSON, err := json.Marshal(tokenReview)
+	if err != nil {
+		klog.Errorf("failed to marshal TokenReview to JSON: %v", err)
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, tokenReviewURL.String(), bytes.NewBuffer(tokenReviewJSON))
```

Comment:
lets extract the call into a separate function or `openShiftAuth` method, which would return `responseTokenReview` and error, so its more obvious and also testable.